src/main/java/edu/greenblitz/tobyDetermined/commands/telescopicArm/moveByNodes/StraightLineMotion.java

```diff
@@ -0,0 +1,146 @@
+    }
+
+    public double calculateExtenderVelocity(double ratio) {
+        double signOfExtender = Math.signum(endNode.getExtendPosition() - extender.getLength());
+        double extenderVelocity = Math.sqrt(combinedVelocity * combinedVelocity / (ratio * ratio + 1));
+        double wantedVelocity = signOfExtender * extenderVelocity;
+
+        wantedVelocity = limit(wantedVelocity, MAX_EXTENDER_VELOCITY);
+
+        return wantedVelocity;
+    }
+
+    public double calculateAngularVelocity(double startVelocity) {
+        double signOfAngle = Math.signum(endNode.getAnglePosition() - elbow.getAngleRadians());
+        double magnitudeOfVelocity = startVelocity / (extender.getLength() + STARTING_LENGTH);
+
+        magnitudeOfVelocity = limit(magnitudeOfVelocity, MAX_ANGULAR_VELOCITY);
+
+        return signOfAngle * Math.abs(magnitudeOfVelocity);
+    }
+
+    public double limit(double v, double maxMagnitude) {
+        return limit(v, -maxMagnitude, maxMagnitude);
+    }
+
+    public double limit(double v, double min, double max) {
+        return Math.min(max, Math.max(min, v));
+    }
+
+    public void moveArm() {
+        double startLength = extender.getLength() + STARTING_LENGTH;
+        double endLength = endNode.getExtendPosition() + STARTING_LENGTH;
+        double gamma = endNode.getAnglePosition() - elbow.getAngleRadians();
+
+        double ratio = getRatioBetweenAngleAndLength(startLength, endLength, gamma);
+
+        double extenderVelocity = calculateExtenderVelocity(ratio);
+        double angularVelocity = calculateAngularVelocity(ratio * extenderVelocity);
+
+        if (!endNode.isAtAngle(elbow.getAngleRadians()))
+            elbow.setMotorVoltage(Elbow.getDynamicFeedForward(angularVelocity, extender.getLength(), elbow.getAngleRadians()));
+        else
+            elbow.setMotorVoltage(Elbow.getStaticFeedForward(extender.getLength(), elbow.getAngleRadians()));
+
+
+        if ((endLength > EXTENDER_STOPPING_THRESHOLD || startLength > EXTENDER_STOPPING_THRESHOLD)
+                && !endNode.isAtLength(extender.getLength()))
+            extender.setMotorVoltage(Extender.getDynamicFeedForward(extenderVelocity, Extender.getDynamicFeedForward(extenderVelocity, Elbow.getInstance().getAngleRadians())));
+        else
+            extender.setMotorVoltage(Extender.getStaticFeedForward(elbow.getAngleRadians()));
+
+    }
+
+    public boolean isInPlace() {
+        return endNode.isAtNode(elbow.getAngleRadians(), extender.getLength());
+    }
+
+    @Override
+    public void execute() {
+        moveArm();
+    }
+
+    @Override
+    public boolean isFinished() {
+        return isInPlace();
+    }
+
+    @Override
+    public void end(boolean interrupted) {
+        if (interrupted) {
+            system1MidNode.setMidNode(startState, endState);
+            system1CurrentNode.setCurrentNode(SystemsState.MID_NODE_1);
+        } else
+            system1CurrentNode.setCurrentNode(endState);
+    }
+}
```

Comment:
File ~could~ should be shorter

Reply:
"The first rule of classes is that they should be small. The second rule of classes is that they should be smaller than that.".